test/junit/scala/collection/immutable/QueueTest.scala

```diff
@@ -37,4 +37,14 @@ class QueueTest {
     assertEquals(0, Queue(1,2).copyToArray(target, 1, 0))
   }
 
+  @Test def `iterator works as lazy concat`(): Unit = {
```

Comment:
It doesn't really test that though. It tests that the iterator returns the correct results.

Reply:
I guess the code comment was that I manually tested it. Thanks for noticing. Also thanks Stefan for the many neat things you do.

Reply:
OK, https://github.com/scala/scala/pull/8605

I should rename this test in user story form: `As a lazily concated iterator, I would like to return correct results.`